### ISSUE TITLE
Replace or drop "cross-fetch" since it is outdated and obsolete

### ISSUE THREAD

**Issue by dirkluijk** [IanVS/vitest-fetch-mock#23]
When using this library with Vitest on Node.js 22.x, I get a lot of annoying warnings:
```
(node:22221) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
```
I traced this to the following dependencies:
```
└─┬ vitest-fetch-mock@0.3.0
  └─┬ cross-fetch@4.0.0
    └─┬ node-fetch@2.7.0
      └── whatwg-url@5.0.0
```

Related downstream issues / MRs:
- https://github.com/lquixada/cross-fetch/pull/183
- https://github.com/lquixada/cross-fetch/issues/177

An easy solution could also to get rid of "cross-fetch" as a dependency, since Fetch is widely available (at least from Node.js 18 and higher). If there is an agreement for this solution, I'll prepare a pull request for this.

**Comment by IanVS** [comment]
Related: https://github.com/IanVS/vitest-fetch-mock/issues/22

**Comment by dirkluijk** [comment]
I did a few hours of hacking and created a small proof-of-concept that seems to be more type-safe but also has no other dependencies other than Vitest.

Please find the Gist here: 
https://gist.github.com/dirkluijk/6f4ddb316a6d7c29e5feab5073ab956f

Maybe this setup can become the basis for a new major version of this library? Let me know, I am willing to collaborate on it.

**Comment by IanVS** [comment]
Hi, that sounds great.  I really haven't done much work on this project other than forking it and making it work with vitest, and I hardly use it for my own projects, so having someone else to help work on it sounds great.  

As far as the other issue I linked to, I suppose as long as we are only ever reading `fetch`, `Response`, etc from `globalThis`, then the user is free to either use built-in fetch or mock out fetch with whatever polyfill they desire, so long as it correctly sets all the globals we need.  Is that your thought as well?  I like the idea of decoupling from a fetching library, it will make this tool much more flexible.

**Comment by dirkluijk** [comment]
Agreed! I will work on a PR. It might be a bit of a refactor, but I’ll document any (breaking) changes.

**Comment by dirkluijk** [comment]
See https://github.com/IanVS/vitest-fetch-mock/pull/24.

**Comment by IanVS** [comment]
Closed by https://github.com/IanVS/vitest-fetch-mock/pull/24, will be released in the next version.
